pkg/controller/seed-controller-manager/kubernetes/resources_test.go

```diff
@@ -215,35 +200,6 @@ func TestCloudControllerManagerDeployment(t *testing.T) {
 			},
 			wantCCMCreator: false,
 		},
-		{
-			name: "No CCM migration ongoing",
-			cluster: &kubermaticv1.Cluster{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "cluster-a",
-				},
-				Spec: kubermaticv1.ClusterSpec{
-					Features: map[string]bool{
-						kubermaticv1.ClusterFeatureExternalCloudProvider: true,
-					},
-				},
-				Status: kubermaticv1.ClusterStatus{
-					NamespaceName: "test",
-				},
-			},
-			kcmDeploymentConfig: KCMDeploymentConfig{
-				Flags: []string{},
-				Status: appsv1.DeploymentStatus{
-					ObservedGeneration: 1,
-					Replicas:           2,
-					AvailableReplicas:  1,
-					UpdatedReplicas:    1,
-				},
-				Generation: 1,
-				Replicas:   2,
-				Namespace:  "test",
-			},
-			wantCCMCreator: true,
-		},
```

Comment:
This test case has been removed intentionally because it's failing, however, it's expected that it'll fail.

We deploy/reconcile the external CCM only if the KCM has finished rolling-out. For example, if we're upgrading the cluster, we wouldn't reconcile the external CCM until the KCM has not finished rolling-out. What's important is that the external CCM will not be removed while it's not reconciled, so this should be okay.